My bowling

junohm410 opened the pull request:
ボウリングスコア計算プログラムを提出します。
レビューのほど、よろしくお願い申し上げます🙇‍♂️

junohm410 committed:
bowling.rbを追加

ボウリングのスコア計算アプリ bowling.rb を追加しました。

junohm410 committed:
修正版bowling.rbをコミットしました

前回のコミットで修正前のプログラムをコミットしたため、修正版をaddしコミットしました。

JunichiIto commented on 03.bowling/bowling.rb:
```diff
@@ -0,0 +1,45 @@
+  end
+end
+
+frames = []
+shots.each_slice(2) do |shot|
+  frames << shot
+end
```
```suggestion
frames = shots.each_slice(2).to_a
```

でも同じ結果になるはずですー

junohm410 replied:
こちら、同じ結果になることを確認いたしました🙇‍♂️
`each_slice`から返ってきた`Enumerator`オブジェクトを`to_a`で配列にできるという流れをリファレンスマニュアルでも確認しました。https://docs.ruby-lang.org/ja/latest/method/Enumerable/i/entries.html
修正しております。https://github.com/junohm410/ruby-practices/pull/3/commits/273d91750cf31deed971df29057ce51af4e3e2a1

JunichiIto commented on 03.bowling/bowling.rb:
```diff
@@ -0,0 +1,45 @@
+  frames << shot
+end
+
+point = 0
+frames.each.with_index do |frame, i|
```
```suggestion
point = frames.each_with_index.sum do |frame, i|
```

のようにsumの結果を直接変数に入れる方法も採れそうです。

junohm410 replied:
こちら、ありがとうございます🙇‍♂️
こちらのコードで動くことを確認しました。
ただ、ここの理解が怪しいため、フィヨルドの提出ページの方で疑問点を質問させていただいております。
そちらをご確認いただけるとありがたいです。宜しくお願いいたします🙏

JunichiIto commented on 03.bowling/bowling.rb:
```diff
@@ -0,0 +1,45 @@
+
+point = 0
+frames.each.with_index do |frame, i|
+  next_frame = i + 1
```
`i + 1`で取れるのはフレームのデータそのものではなく、ただの添え字（数字）ですね。
であれば`next_frame_index`のような名前の方がいいと思うのですが、そもそも添え字ではなくフレームデータを以下のように代入した方がコードがスッキリしそうです。

```suggestion
  next_frame = frame[i + 1]
```

junohm410 replied:
ご指摘ありがとうございます🙇‍♂️
2点、確認しました。
- `i + 1`で取れるのはただの添え字なので、私の最初のコードの変数の命名は不親切。
- フレームデータ（個々の配列）をそもそも代入していった方がコードがすっきりする。

後者の形でループ内の処理を修正しました。https://github.com/junohm410/ruby-practices/pull/3/commits/273d91750cf31deed971df29057ce51af4e3e2a1

JunichiIto reviewed (commented):
コメントしました〜。

junohm410 committed:
レビューで指摘いただいた点を修正

frames配列・変数pointsへの値の代入方法と、next_frame,frame_after_nextに代入する対象に関し、レビューを受け修正いたしました。

junohm410 commented:
レビューで指摘いただいた点を修正いたしました。
ご確認のほど、何卒よろしくお願いいたします🙇‍♂️

JunichiIto commented on 03.bowling/bowling.rb:
```diff
@@ -0,0 +1,40 @@
+  if i < 9
+    if frame[0] == 10
+      if next_frame[0] == 10
+        10 + frame_after_next[0] + frame.sum
```
この10は`next_frame[0]`の10だと思います。
ここだけリテラルの10が登場するのは一貫性がないので、

```suggestion
        next_frame[0] + frame_after_next[0] + frame.sum
```

みたいになってる方がいいかなと思いました。（frame, next_frame, frame_after_next、の順で並ぶとより自然かも？）

junohm410 replied:
ストライクだからまずは10を足して……という感じで書いていましたが、確かに一貫性がなく可読性が低いと思いました🙇‍♂️
修正し、ループ内でのスコアの加算の順番も frame, next_frame, frame_after_next になるよう、その他の箇所も修正しております。https://github.com/junohm410/ruby-practices/pull/3/commits/f6ff1b75f8b03f05e7bbe4e9f1d8e92e3edf4ea2

JunichiIto commented on 03.bowling/bowling.rb:
```diff
@@ -0,0 +1,40 @@
+
+frames = shots.each_slice(2).to_a
+
+point = frames.each.with_index.sum do |frame, i|
```
```suggestion
point = frames.each.with_index.sum do |current_frame, i|
```

「現在の」を強調してあげると、処理中の`frame`の役割がより明確になりそうです。

JunichiIto replied:
`each.with_index`を`each_with_index`にするとメソッドの呼び出しが1回減らせますね。

junohm410 replied:
`current_frame`にした方がループ内で行われいることの意味が捉えやすいということ、その通りだと思いました🙇‍♂️
また後者のご指摘についても、`each.with_index`で引数を取って添字を変えているわけでもないので、この観点でも`each_with_index`が合理的だと思いました🙇‍♂️修正しております。https://github.com/junohm410/ruby-practices/pull/3/commits/f6ff1b75f8b03f05e7bbe4e9f1d8e92e3edf4ea2

JunichiIto commented on 03.bowling/bowling.rb:
```diff
@@ -0,0 +1,40 @@
+
+point = frames.each.with_index.sum do |frame, i|
+  next_frame = frames[i + 1]
+  frame_after_next = frames[i + 2]
```
```suggestion
  after_next_frame = frames[i + 2]
```

の方が英語的に自然かもしれません

junohm410 replied:
はい、next_frameのことも考えるとafter_next_frameがいいですね……🙇‍♂️ありがとうございます。修正いたしました。https://github.com/junohm410/ruby-practices/pull/3/commits/f6ff1b75f8b03f05e7bbe4e9f1d8e92e3edf4ea2

JunichiIto reviewed (commented):
コメントしました〜。
質問の件はFBCのコメント欄で返信しますね。

junohm410 committed:
2回目のレビュー内容を受けプログラムを修正

each内の可読性を上げるために変数名や加算の並びを変更するなど、レビュー内容を反映し修正いたしました。

junohm410 commented:
レビューとFBCでのご回答、誠にありがとうございました🙇‍♂️修正いたしました。
ご確認のほど、宜しくお願い申し上げます。

JunichiIto reviewed (approved):
確認しました！OKです〜🙆‍♂️

junohm410 commented:
ご確認、ありがとうございました🙇‍♂️
